### PR TITLE
SOV-1086: feat: helper functions for working with ethers bignumbers

### PR DESCRIPTION
**Pull request by creed-victor**
https://sovryn.atlassian.net/browse/SOV-1086

### CODE REVIEW

File: apps/frontend/src/utils/math.test.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  it('returns gas in gwei', () => {
+    const value = formatUnits(
+      parseUnits(gasPrice.toString(), 'gwei').mul(gasLimit),
+      'gwei',
+    );
+    expect(value).toBe('3079.259656');
+  });
+
+  it('returns gas in gwei', () => {
```
Comment (tiltom) [inline apps/frontend/src/utils/math.test.ts]: You have 2 identical tests.

File: apps/frontend/src/utils/math.test.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    const value = formatUnits(
+      parseUnits(gasPrice.toString(), 'gwei').mul(gasLimit),
+      'gwei',
+    );
+    expect(value).toBe('3079.259656');
+  });
+
+  it('adds comma as thousand separator', () => {
```
Comment (tiltom) [inline apps/frontend/src/utils/math.test.ts]:
```suggestion
  it('adds comma as thousands separator', () => {
```

File: apps/frontend/src/utils/math.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { isBigNumberish } from '@ethersproject/bignumber/lib/bignumber';
+
+import { BigNumberish, BigNumber } from 'ethers';
+import { formatUnits, parseUnits } from 'ethers/lib/utils';
+
+// helper function to convert any type of ethers value to wei.
+export const toWei = (value: BigNumberish): BigNumber => {
+  if (isBigNumberish(value)) {
+    return BigNumber.from(value).mul(BigNumber.from(10).pow(18));
```
Comment (tiltom) [inline apps/frontend/src/utils/math.ts]: I think that this depends on a number of decimals, right? I know that 18 is the most common but what if we have a token with a different amount of decimals?

**Review by tiltom — CHANGES_REQUESTED** [review]

File: apps/frontend/src/utils/math.ts (hunk 3 above)
Comment (creed-victor) [reply apps/frontend/src/utils/math.ts]: good point, will add argument for overriding decimals points.

File: apps/frontend/src/utils/math.test.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+import { commify, formatUnits, parseUnits } from 'ethers/lib/utils';
+
+import { fromWei, fromWeiFixed, toWei } from './math';
+
+const gasPrice = 0.065164;
+const gasLimit = 47254;
+
+describe('utils/math.ts', () => {
+  describe('toWei()', () => {
+    it('converts floating numbers to wei', () => {
```
Comment (tiltom) [inline apps/frontend/src/utils/math.test.ts:11]: Rewrite to one-liner pls.

File: apps/frontend/src/utils/math.test.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    });
+
+    it('converts gwei floating numbers to wei', () => {
+      const value = toWei(gasPrice)
+        .div(10 ** 9)
+        .toString();
+      expect(value).toEqual('65164000');
+    });
+
+    it('converts full number to wei', () => {
```
Comment (tiltom) [inline apps/frontend/src/utils/math.test.ts:22]: Rewrite to one-liner pls.

File: apps/frontend/src/utils/math.test.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+  it('returns gas in gwei', () => {
+    const value = formatUnits(
+      parseUnits(gasPrice.toString(), 'gwei').mul(gasLimit),
+      'gwei',
+    );
+    expect(value).toBe('3079.259656');
+  });
+
+  it('adds comma as thousands separator', () => {
```
Comment (tiltom) [inline apps/frontend/src/utils/math.test.ts:130]: Rewrite to one-liner pls.

File: apps/frontend/src/utils/math.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import { isBigNumberish } from '@ethersproject/bignumber/lib/bignumber';
+
+import { BigNumberish, BigNumber } from 'ethers';
+import { formatUnits, parseUnits } from 'ethers/lib/utils';
+
+const DEFAULT_UNIT = 18;
+const DEFAULT_DECIMALS = 6;
+
+const unitNames = ['wei', 'kwei', 'mwei', 'gwei', 'szabo', 'finney', 'ether'];
```
Comment (tiltom) [inline apps/frontend/src/utils/math.ts:9]:
```suggestion
const units = {
  wei: 0,
  kwei: 3,
  mwei: 6,
  gwei: 9,
  szabo: 12,
  finney: 15,
  ether: 18,
};
```
I'd rather have an object with mapping than just an array.

File: apps/frontend/src/utils/math.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+export const toWei = (
+  value: BigNumberish,
+  unitName: BigNumberish = DEFAULT_UNIT,
+): BigNumber => {
+  if (isBigNumberish(value)) {
+    if (typeof unitName === 'string') {
+      const index = unitNames.indexOf(unitName);
+      if (index !== -1) {
+        unitName = 3 * index;
+      }
```
Comment (tiltom) [inline apps/frontend/src/utils/math.ts:21]:
```suggestion
unitName = units[unitName];
```
I think this is a better approach compared to multiplying an index from the `unitNames` array with a seemingly random number (`3`).

**Review by tiltom — CHANGES_REQUESTED** [review]

File: apps/frontend/src/utils/math.ts (hunk 7 above)
Comment (creed-victor) [reply apps/frontend/src/utils/math.ts:9]: ethers bignumber accepts units as it's defined here so I prefer to use it same way to be fully comply with it.

Comment (tiltom) [reply apps/frontend/src/utils/math.ts:9]: ok then, let's leave it that way

**Review by tiltom — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Pull request merged**